include/s3select.h

```diff
@@ -274,6 +275,30 @@ struct push_debug_1
 };
 static push_debug_1 g_push_debug_1;
 
+struct push_trim_type
+{
+  void operator()(s3select* self, const char* a, const char* b) const;
+};
+static push_trim_type g_push_trim_type; 
+
+struct push_trim_expr_one
```

Comment:
rename push_trim_expr_one/two/three to something meaningful (leading,trailing,both)